packages/config/src/handlers/updateFormDraft/handler.ts

```diff
@@ -0,0 +1,186 @@
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import Question, {
+  IQuestion,
+  IQuestionModel,
+  validFieldType
+} from '@config/models/question'
+import * as Hapi from '@hapi/hapi'
+import { logger } from '@config/config/logger'
+import * as Joi from 'joi'
+import { internal } from '@hapi/boom'
+import FormDraft, {
+  IFormDraft,
+  IHistory,
+  IFormDraftModel,
+  validStatus,
+  validEvent,
+  DraftStatus
+} from '@config/models/formDraft'
+import { messageDescriptorSchema } from '@config/handlers/createQuestion/handler'
+
+export interface IQuestionsDraft extends IFormDraftModel {
+  questions?: IQuestion[]
+  //deleted contains list of fieldId to delete
+  deleted?: string[]
+}
+
+export async function updateFormDraftHandler(
+  request: Hapi.Request,
+  h: Hapi.ResponseToolkit
+) {
+  const questionsDraft = request.payload as IQuestionsDraft
+
+  if (
+    !questionsDraft.questions &&
+    !questionsDraft.deleted &&
+    questionsDraft.status === DraftStatus.DRAFT
+  ) {
+    return h.response('Nothing to add or remove from Question').code(400)
+  }
+
+  let draft: IFormDraftModel | null = await FormDraft.findOne({
+    event: questionsDraft.event
+  })
+
+  if (draft) {
+    //update draft
+    const history: IHistory = {
+      version: draft.version,
+      status: draft.status,
+      comment: draft.comment,
+      lastUpdateAt: draft.updatedAt
+    }
+
+    draft.history?.unshift(history)
+    draft.status = questionsDraft.status
+    draft.comment = questionsDraft.comment
+    draft.version =
+      draft.status === DraftStatus.PUBLISHED ||
+      draft.status === DraftStatus.PREVIEW ||
+      draft.status === DraftStatus.FINALISED
+        ? draft.version
+        : draft.version + 1
+    draft.updatedAt = Date.now()
+
+    try {
+      await FormDraft.update({ _id: draft._id }, draft)
+    } catch (err) {
+      logger.error(err)
+      return h
+        .response(`Could not update draft for ${draft.event} event`)
+        .code(400)
+    }
+  } else {
+    //create draft
+    try {
+      const formDraft: IFormDraft = {
+        event: questionsDraft.event,
+        status: questionsDraft.status,
+        comment: questionsDraft.comment,
+        version: 1,
+        createdAt: Date.now(),
+        updatedAt: Date.now()
+      }
+      draft = await FormDraft.create(formDraft)
+    } catch (e) {
+      throw internal(e.message)
+    }
+  }
+
+  //update existing questions
+  if (questionsDraft.questions) {
+    const existingQuestions: IQuestionModel[] = await Question.find().exec()
+
+    questionsDraft.questions.forEach(async (question) => {
+      const existingQuestion = existingQuestions.find(
+        (extQuestion) => extQuestion.fieldId === question.fieldId
```

Comment:
@euanmillar as you were saying in the previous sprint planning, would it be better to keep a unique id in the `Question` schema?

Reply:
Question schema already has _id but since fieldId also must be unique so it is better to use fieldId for the filter.